lib/spack/spack/spec.py

```diff
@@ -887,6 +989,13 @@ def concretized(self):
 
 
     def flat_dependencies(self, **kwargs):
+        flat_deps = DependencyMap()
+        for name, depspec in self.flat_dependencies_with_deptype(**kwargs).items():
+            flat_deps[name] = depspec.spec
+        return flat_deps
+
+
+    def flat_dependencies_with_deptype(self, **kwargs):
```

Comment:
How about a `_full` suffix? `_depspec`?

Not sure about that case, would have to test. It might prefer only the first deptype set it comes across (cf. `_dup`).